src/tqec/templates/library/spatial.py

```diff
@@ -0,0 +1,413 @@
+    # plaquette should be removed from the mapping (this is the case where it
+    # has been set twice in the ifs above).
+    if arms == JunctionArms.UP | JunctionArms.LEFT:
+        del mapping[4]
+    elif arms == JunctionArms.DOWN | JunctionArms.RIGHT:
+        del mapping[1]
+
+    ####################
+    #       Bulk       #
+    ####################
+    # Assigning plaquette description to the bulk, considering that the bulk
+    # corners (i.e. indices {5, 6, 7, 8}) should be assigned "regular" plaquettes
+    # (i.e. 6 and 7 have the same plaquette as 17, 5 has the same plaquette as
+    # 13 and 8 has the same plaquette as 15). If these need to be changed, it
+    # will be done afterwards.
+    internal_basis_plaquette = RPNGDescription.from_string(
+        f"{r}{bi}1{m} {r}{bi}3{m} {r}{bi}2{m} {r}{bi}5{m}"
+    )
+    mapping[6] = internal_basis_plaquette
+    mapping[7] = internal_basis_plaquette
+    mapping[17] = internal_basis_plaquette
+
+    # be{h,v}hp: basis external {horizontal,vertical} hook plaquette
+    behhp = RPNGDescription.from_string(
+        f"{r}{be}1{m} {r}{be}2{m} {r}{be}3{m} {r}{be}4{m}"
+    )
+    bevhp = RPNGDescription.from_string(
+        f"{r}{be}1{m} {r}{be}4{m} {r}{be}3{m} {r}{be}5{m}"
+    )
+    mapping[5] = bevhp if JunctionArms.UP in arms else behhp
+    mapping[13] = bevhp if JunctionArms.UP in arms else behhp
+    mapping[14] = behhp if JunctionArms.RIGHT in arms else bevhp
+    mapping[8] = bevhp if JunctionArms.DOWN in arms else behhp
+    mapping[15] = bevhp if JunctionArms.DOWN in arms else behhp
+    mapping[16] = behhp if JunctionArms.LEFT in arms else bevhp
+
+    # In the special cases of an L-shaped junction TOP/LEFT or DOWN/RIGHT, the
+    # opposite corner **within the bulk** should be overwritten to become a
+    # 3-body stabilizer measurement.
+    if arms == JunctionArms.UP | JunctionArms.LEFT:
+        mapping[8] = RPNGDescription.from_string(
+            f"{r}{be}1{m} {r}{be}2{m} {r}{be}4{m} ----"
+        )
+    elif arms == JunctionArms.DOWN | JunctionArms.RIGHT:
+        mapping[5] = RPNGDescription.from_string(
+            f"---- {r}{be}2{m} {r}{be}4{m} {r}{be}5{m}"
+        )
+
+    ####################
+    #  Sanity checks   #
+    ####################
+    # All the plaquettes in the bulk should be set.
+    bulk_plaquette_indices = {5, 6, 7, 8, 13, 14, 15, 16, 17}
+    missing_bulk_plaquette_indices = bulk_plaquette_indices - mapping.keys()
+    assert not missing_bulk_plaquette_indices, (
+        "Some plaquette(s) in the bulk were not correctly assigned to a "
+        f"RPNGDescription. Missing indices: {missing_bulk_plaquette_indices}."
+    )
+
+    return RPNGTemplate(
+        template=QubitSpatialJunctionTemplate(),
+        mapping=FrozenDefaultDict(
+            mapping,
+            default_factory=lambda: RPNGDescription.from_string("---- ---- ---- ----"),
+        ),
+    )
+
+
+def get_spatial_junction_junction_template(
```

Comment:
```suggestion
def get_spatial_junction_arm_template(
```
Some terminology in the documentation also needs to be revised.